### ISSUE TITLE
get_attribute_hook does not trigger for class attributes, get_base_class_hook cannot resolve class attr types

### ISSUE THREAD

**Issue by FuegoFro** [python/mypy#9645]
**Bug Report**

I wasn't sure how to best phrase title and/or description for this, so edits/suggestions/renames are welcomed.

The `get_attribute_hook` plugin hook is not triggered for attributes/variables on classes. It seems to only trigger for attributes/variables on instances. Additionally, I seem to be unable to resolve types of class variables in the hook return by `get_base_class_hook`.

**To Reproduce**

Given the source file

```python
from typing import Dict, TYPE_CHECKING, Tuple, cast


class Field:
    pass


class FieldBuilder:
    def with_name(self) -> 'FieldBuilder':
        return self

    # Imagine arbitrarily more builder methods

    def build(self) -> Field:
        return Field()


class BaseMeta(type):
    def __new__(mcs, name: str, bases: Tuple[type, ...], attrs: Dict[str, object]) -> 'BaseMeta':
        for key, val in attrs.items():
            if isinstance(val, FieldBuilder):
                attrs[key] = val.build()
        return cast(BaseMeta, type.__new__(mcs, name, bases, attrs))


class Base(metaclass=BaseMeta):
    pass


class Concrete(Base):
    foo = FieldBuilder().with_name()  # Can have arbitrary chain of builder methods
    bar = 1


if TYPE_CHECKING:
    reveal_type(Concrete.foo)  # Would like this to show up as type "Field"
    reveal_type(Concrete.bar)  # Would like this to show up as type "int"
print(type(Concrete.foo))
print(type(Concrete.bar))
```

and the plugin

```python
from typing import Callable, Optional
from typing import Type as TypingType

from mypy.plugin import AttributeContext, ClassDefContext, Plugin
from mypy.types import Type as MypyType


class ExamplePlugin(Plugin):
    def get_attribute_hook(self, fullname: str) -> Optional[Callable[[AttributeContext], MypyType]]:
        print('ATTRIBUTE HOOK FOR', fullname)
        return None

    def get_base_class_hook(self, fullname: str) -> Optional[Callable[[ClassDefContext], None]]:
        if not fullname.endswith('.Base'):
            return None

        def hook(ctx: ClassDefContext) -> None:
            print('BASE CLASS HOOK FOR', fullname, ctx.cls.fullname)
            if ctx.cls.info.names['foo'].type is None:
                if ctx.api.final_iteration:
                    print('CANNOT DEFER ANY MORE', fullname)
                else:
                    print('DEFERRING', fullname)
                    ctx.api.defer()

        return hook

def plugin(_mypy_info: str) -> TypingType[Plugin]:
    return ExamplePlugin
```

**Expected Behavior**

I would expect that an `ATTRIBUTE HOOK FOR` line would be printed for the `Concrete.foo` and `Concrete.bar` accesses. Alternatively, I'd expect to be able to eventually get a non-`None` type for the `foo` attribute in the base class hook.

**Actual Behavior**

The `ATTRIBUTE HOOK FOR` is never printed for the `Concrete.*` accesses and we never get a non-`None` type for `foo` (we reach the final iteration).

If there's a way to accomplish the `Would like this to show up as type` comments with the existing plugin hook behaviors, that also works for me! Note, I've seen [other plugins](https://github.com/dropbox/sqlalchemy-stubs/blob/55470ceab8149db983411d5c094c9fe16343c58b/sqlmypy.py#L79-L91) use the `get_function_hook` to modify the return type and therefore the type of the class var, but I didn't see a way to know that the result of the function would be stored in a class var, since I wouldn't want *every* builder method to be automatically converted.

**Your Environment**

- Mypy version used: `mypy 0.800+dev.f220ce50725ed7f117832b9f03aed4d1a9508e00`
- Mypy command-line flags: `--show-traceback --warn-unused-configs --config-file <file>`
- Mypy configuration options from `mypy.ini` (and other config files):
```ini
[mypy]
plugins = <my plugin>
strict = True
```
- Python version used: `3.7.6`
- Operating system and version: macOS 10.15.7
